docstore/awsdynamodb/codec.go

```diff
@@ -260,9 +260,9 @@ func (d decoder) MapLen() (int, bool) {
 	return len(d.av.M), true
 }
 
-func (d decoder) DecodeMap(f func(key string, vd driver.Decoder) bool) {
+func (d decoder) DecodeMap(f func(key string, vd driver.Decoder, exactMatch bool) bool) {
 	for k, av := range d.av.M {
-		if !f(k, decoder{av}) {
+		if !f(k, decoder{av}, true) {
```

Comment:
I'm sorry, I don't understand. How does this `DecodeMap` method know what value of `exactMatch` is passed to it? It shouldn't care, it should follow its contract. (You and I know that it will never be passed `false`, but it shouldn't make that decision for us.)

Reply:
Oh, sorry, I didn't explain myself well. I mean this part is the body of the function `DecodeMap`, not the anonymous function of argument `f`. I am passing `true` to `f` as `f` needs to take in an argument `exactMatch`. The function body of `f` is actually in driver/codec.go [here](https://github.com/google/go-cloud/blob/master/docstore/driver/codec.go#L657)

Reply:
Oh of course, stupid of me: I read `exactMatch` as an arg to `DecodeMap`, but it isn't.